pkg/daemon/daemon.go

```diff
@@ -545,15 +548,17 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 		glog.Errorf("nodeStateSyncHandler(): fail to restart device plugin pod: %v", err)
 		return err
 	}
-	if anno, ok := dn.node.Annotations[annoKey]; ok && (anno == annoDraining || anno == annoMcpPaused) {
+	if dn.isNodeDraining() {
 		if err := dn.completeDrain(); err != nil {
 			glog.Errorf("nodeStateSyncHandler(): failed to complete draining: %v", err)
 			return err
 		}
-	} else if !ok {
-		if err := dn.annotateNode(dn.name, annoIdle); err != nil {
-			glog.Errorf("nodeStateSyncHandler(): failed to annotate node: %v", err)
-			return err
+	} else {
+		if !dn.nodeHasAnnotation(annoKey, annoIdle) {
```

Comment:
> but later on we might have additional state. thus, i think we should set annotation to idle only for case-1 only.
then if later on additional values are added this would not automatically reset annotation to idle.

i assume that if additional values are added it would need to be handled in other places as well. so its not an issue with this PR, im probably overthinking this :\

current approach is fine with me